drivers/watchdog/wdt_mchp_xec.h

```diff
@@ -0,0 +1,47 @@
+/* wdt_xec.h - Microchip XEC watchdog driver private definitions */
+
+/*
+ * Copyright (c) 2019 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_WDT_WDT_XEC_H_
+#define ZEPHYR_DRIVERS_WDT_WDT_XEC_H_
+
+struct wdt_xec_data {
+	wdt_callback_t cb;
+	bool timeout_installed;
+};
+
+/* See datasheet Section 18.7 */
+
+struct wdt_xec_registers {
```

Comment:
Using a HAL is up to the driver maintainer. And actually, the actual HAL is not really a HAL. It does not provide much. If you compare it to ST's or Atmel's ... this one is pretty useless.
It would provide low level functions to deal with the device, hiding the actual nature of the regs, then yes I would use. But it does not.

Due to this, it brings a huge limitation: porting the driver to other MEC's which own about the same exact IP block. Let's take mec1701 for instance: 95% same block (it just does not have interrupt), but different "HAL"... so it means a new driver for it?
With my driver it will be just 2 or 3 ifdefs at the right place, and done.

I am going to keep it that way.

There is one thing which I am using from the HAL is the EC enablement of the interrupt, but I think it should not be there (I just did it because other xec driver do, and I am obviously now in a hurry to get it merged). It should be in SoC. EC part of MEC is really specific and all of its logic should be in soc.h/c part. Will see if we can make it better someday.

Reply:
The MEC1701 is something we used to start exploring  Zephyr  as there wasn't any clear release date for the MEC1501 back then. We don't care about the MEC1701.